keps/sig-storage/20190122-csi-inline-volumes.md

```diff
@@ -198,63 +120,28 @@ type CSIVolumeSource struct {
 	// Required.
 	Driver string
 
-	// VolumeHandle is the unique volume name returned by the CSI volume
-	// plugin’s CreateVolume to refer to the volume on all subsequent calls.
-	// If not provided, that handle will be auto-generated.
-	//
-	// +optional
-	VolumeHandle *string
-
-	// ReadOnly is the value to pass to ControllerPublishVolumeRequest.
-	// Defaults to false (read/write).
-	// 
-	// +optional
-	ReadOnly *bool
-
-	// Filesystem type to mount. Ex. "ext4", "xfs", "ntfs".
-	// If not provided, the empty value is passed to the associated CSI driver
-	// which will determine the default filesystem to apply.
-	//
-	// +optional
-	FSType *string
```

Comment:
isn't fstype used at mount time? is this not relevant for a purely ephemeral volume?

Reply:
It depends on if the ephemeral volume will actually create/attach a separate block device that would need to be formatted.  Right now the use cases I've heard of (secret/configmap-like, container images, tar files, etc) would just utilize the root disk.

Reply:
optional wouldn't hurt I guess. You could have an ephemeral driver that, say, provisions lvm volumes temporarily, formats them with FSType, and then releases the volume when the pod dies.

Size may be relevant then too.

Reply:
I don't want to add size here yet. That potentially implies that we need things like capacity reporting and scheduling support, which I think we should consider at a later time.

Reply:
Discussed with @saad-ali. Until we have someone come to us with a use case for needing fstype, let's leave it out of the API.

Reply:
pmem-csi (https://github.com/intel/pmem-CSI) manages local PMEM storage. It needs both size and fstype because an ephemeral volume will be a block device that gets allocated from the locally available PMEM.

I agree that this raises questions around whether an ephemeral volume really can be allocated on the node that a pod was assigned to. I mentioned this myself in an earlier review of the inline volume KEP. But just because those questions don't have a good (or rather, any) answer doesn't mean that it should be impossible to request such blocks. Users simply have to be aware that the pod may fail to start. For an alpha feature that's okay.

As I mentioned earlier, trying to unpack a tar file or download a container image file can also fail. That particular failure mode isn't specific to drivers that allocate a new block device.

Reply:
The CSI spec has no notion of size during NodePublish (mount).  The `VolumeAttributes` can be used to send non-standard parameters (such as `size` or anything else) to the driver.

Reply:
Thanks for bringing up that use case.  I'm ok with adding fstype back since CSI drivers are supposed to be passed the fstype by the CO in the volume capabilities.  

However, I don't think we add a size field until we have a detailed design on how the overall ephemeral inline csi volume feature will integrate with the [local ephemeral storage management](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#local-ephemeral-storage) feature.  That design may impact what we want to put in the API (support both requests + limits or just one of them, etc).  cc @jingxu97.  As a temporary workaround for alpha, drivers can accept a size attribute in the volume context.

Reply:
Let's revisit this prior to Beta to figure which fields should be surfaced as first class parameters.

Reply:
Agreed.